### PR TITLE
update scale_bar example to include new attributes

### LINKED ISSUE #7007: Update the scale bar example for the latest API
## 🧰 Task
The scale bar example
https://napari.org/dev/gallery/scale_bar.html

Doesn't cover the latest scale bar API, which permits a lot more.
It should be updated to include some of the new options:
https://github.com/napari/napari/blob/0e01840c9237489acf238aec1d5cdaf9201ed640/napari/components/overlays/scale_bar.py#L10-L46

Would be good to also update the docs, see:
https://github.com/napari/docs/issues/38

Motivation:
User post on image.sc 
https://forum.image.sc/t/increasing-font-size-and-repositioning-the-scale-bar/97972/3

### PR DESCRIPTION
**Pull request by TimMonko**
# References and relevant issues
Closes #7007

# Description
Adds various attributes to scale_bar example to show some options, including the typical settings of setting the unit, desired length, and increasing the font size. 

Known visual issues with this scale bar example include enlarged text clipping outside `scale_bar.box` (issue #5914) and moving the `scale_bar.position` to anywhere on the top clips outside the viewer (PR #7018). These visual bugs are left in to show the user some of the limitations. 

Trying to work on some of these issues to learn the napari library better, seems to be the point of the 'good first issue' tag, but please feel free to give extra guidance.

### CODE REVIEW

File: examples/scale_bar.py
Comment (jni) [inline examples/scale_bar.py]: Minor nitpick: PEP-8 [specifies](https://pep8.org/#inline-comments) two spaces after the code for inline statements:

> Inline comments should be separated by at least two spaces from the statement.

File: examples/scale_bar.py
Comment (jni) [inline examples/scale_bar.py]: Can we use black here? I think almost every other color is distracting here. 😂 🙏 
```suggestion
viewer.scale_bar.box_color = (0, 0, 0, 0.4)  # black with alpha=0.4
```

**Review by jni — APPROVED** [review]
Yes indeed that's what good first issues are for! Thanks for your first PR @TimMonko! 🙏

I've left some minor nitpicks but otherwise this can go in any time as far as I'm concerned! Thanks!

File: examples/scale_bar.py
Comment (TimMonko) [reply examples/scale_bar.py]: Of course, I intentionally chose a rough color so other people would give an opinion!

File: examples/scale_bar.py
Comment (TimMonko) [reply examples/scale_bar.py]: This encourages me to double check I have linting set up correctly in that environment, thank you!

File: examples/scale_bar.py
Comment (psobolewskiPhD) [reply examples/scale_bar.py]: Yeah, I obviously don't think anyone would use magenta for actual display, but I think that a strong color makes it more obvious what is happening, which is the point of the example.

Comment (jni) [reply examples/scale_bar.py]: I think that users can match the black box color to the black box, and anything non-default is good here. In general, as a philosophy I think we should aim for our gallery examples to be pretty, which encourages folks to use the library.

(The case of the cropped text is a bit different, because it shows them that there's a bug, which is indeed important knowledge that you can't get without seeing it. But the box colour is pretty obvious either way?)

**Comment by TimMonko** [comment]
So, I updated the file to include the commenting spaces and changing the box color. I'm a bit confused how the 'reviewed' code works. It looks like it's not actually added to the file until a new commit is made? But it also looks like you can directly accept reviewed codes as commits? If so, I'm wondering why @jni 'approved the changes' if we still had a few things to tease out.

File: examples/scale_bar.py
Comment (TimMonko) [reply examples/scale_bar.py]: I changed the color from magenta to cyan and lowered the alpha to 0.1 so it looks less egregious. I think a black box won't be very obvious given that most of the scale_bar resides outside the image in the example viewer. This way, a slight color can show the clipping issue and what the box is actually doing.

**Comment by DragaDoncila** [comment]
Hi @TimMonko thanks for the PR, and welcome to the repo!

>  It looks like it's not actually added to the file until a new commit is made? But it also looks like you can directly accept reviewed codes as commits?

That's right, review comments with `suggestions` in them can be accepted directly by clicking the `Commit suggestion`. This generates a new commit with the changes, authored by the author of the comment. You can also add many suggestions to a batch and commit at once. 

>  If so, I'm wondering why @jni 'approved the changes' if we still had a few things to tease out.

Good question! I think because the suggestions were mostly nitpicks, Juan was happy to approve the PR knowing that you'd address the few tiny details without him needing to re-review. He also probably assumed that other folks (like me!) would come along and have a look.

Note that a PR in our repo isn't merged until it gets the "ready to merge" label, so an approval with some minor suggestions is quite common.

**Review by DragaDoncila — APPROVED** [review]
All looks good to me! I have no particular issues with the color selection 😆 Once CI is green I'll add ready to merge!

**Review by psobolewskiPhD — APPROVED** [review]
As the one that opened the issue: love it!

...and boy do I have to figure out those vispy text handlings to get that one PR over the line, huh? 😬

**Pull request merged**
